src/background.js

```diff
@@ -360,7 +405,30 @@ const rollout = {
 const setup = {
   async start() {
     const isAddonDisabled = await rollout.getSetting("doh-rollout.disable-heuristics", false);
-    const runAddon = await browser.experiments.preferences.getBoolPref("doh-rollout.enabled", false);
+    const runAddonPref = await browser.experiments.preferences.getBoolPref("doh-rollout.enabled", false);
+    const runAddonBypassPref = await browser.experiments.preferences.getBoolPref("doh-rollout.self-enabled", false);
+    const runAddonLocalStorage = await rollout.getSetting("doh-rollout.doorhanger-decision", false);
+    const remoteDisableAddon = await browser.experiments.preferences.getBoolPref("doh-rollout.remote-disable", false);
+
+    if (remoteDisableAddon) {
+      // Fail safe to remotely disable add-on.
+      const addonPrefs = [
+        "doh-rollout.doorhanger-decision",
+        "doh-rollout.doorhanger-shown",
+        "doh-rollout.previous.trr.mode",
+        "doh-rollout.enabled",
+        DOH_SELF_ENABLED_PREF
```

Comment:
I see that. As long as it's okay/good practice to leave breadcrumb prefs like this. My intent was to clean up after the add-on as best as we can.

Reply:
Prefs are cheap and can be cleaned up. For studies and off-the-trains rollouts, I think it's pragmatic to leave them around. But we should make sure we keep track of what we're leaving behind and clean up when we are done with it.

Note: namespacing is especially important for these so that we don't forget and inadvertently re-use a name later.

Reply:
@nhnt11 I removed the cleanup step!